### DISCUSSION TITLE
How do i upload a folder that contains all my code files to my Github repo using Github's REST API programmatically?

### DISCUSSION THREAD

**Discussion by xts-bit** [community/community#83252] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

How do I upload a folder that contains all my code files to my Github repo using Github's REST API programmatically? Can anyone help me do that in JavaScript with Node.js?

Currently, when I try to run my code, it only uploads files, but how do I upload a folder instead? I don't want to upload files one by one. Any idea what to do?
```

async function upload() {
  const message = 'amogus';
  const content = 'aaaaa';
  const owner = 'MY-USERNAME';
  const repo = 'MY-REPO-NAME';
  const path = 'my-project';
  const auth = '{MY ACCESS TOKEN HERE}';

  const existingFile = await (await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
    {
      method: 'GET',
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${auth}`
      }
    }
  )).json();

  await (await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
    {
      method: 'PUT',
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${auth}`
      },
      body: JSON.stringify({
        message: message,
        content: btoa(content),
        sha: existingFile.sha,
      }),
    }
  )).json();
}

```

**Comment by DozenLimeStone** [comment]
Probably you have to loop through all files in the folder and upload them one by one.

**Reply by xts-bit** [reply]
@DozenLimeStone What if the project has multiple folders with multiple files?

**Reply by DozenLimeStone** [reply, marked as answer]
@xts-bit Well then recursively get them all :) The best other thing you can do is to compress them into one file and send it.

**Reply by xts-bit** [reply]
@DozenLimeStone Yep, So you mean I should first create a .zip file of my project convert it into base64 string then upload it?

**Reply by DozenLimeStone** [reply]
@xts-bit Yes, I mean that. If there is nobody to see the contents, and used to store the data, yes.

**Reply by xts-bit** [reply]
@DozenLimeStone What do you mean by. "If there is nobody to see the contents, and used to store the data" I just tried to upload .zip file but it only uploaded that file with a base64 string like "YW1vZ3VzLmh0bWw=" instead of my code files

any idea?

**Reply by DozenLimeStone** [reply]
@xts-bit  You probably should just read the file in binary, then convert contents into base64 and send.

**Reply by tx0c** [reply]
> compress them into one file and send it.

this is not an answer, because github is the best place to show original code file in its raw format, all programming language code syntax highlighting features, this thread is asking for a solution on how to transfer some potentially huge number of small files in less number of API interactions, ideally 1 API call if possible; see isaacs/github#199 and #88796, doing all files 1 by 1 would cause huge number of API interactions somewhere getting rejected and flagged as API abuse

https://docs.github.com/en/rest/gists/gists?apiVersion=2022-11-28#create-a-gist something like this 1 API call to create gist with multiple files content; can Github contents API be extended to upload multiple files in a single call?

if just compress into a tarball and share, there are so many storage service on the Internet, like Dropbox, GDrive, whatever, why would need Github to share a tarball

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by xts-bit** [comment]
@DozenLimeStone I did this but then it created a myfoldername.zip file in my GitHub repo with a base64 string like "YW1vZ3VzLmh0bWw=" instead of my code files my code: 
```
import fs from 'fs';

function readFileAndConvertToBase64(filePath) {
  try {
    const fileBuffer = fs.readFileSync(filePath);
    const base64Data = fileBuffer.toString('base64');
    return base64Data;
  } catch (error) {
    console.error(error);
  }
}

async function upload() {
  const message = 'hi';
  const content = readFileAndConvertToBase64('./folder.zip');
  const owner = 'MY-USERNAME';
  const repo = 'MY-REPO-NAME';
  const path = 'my-project';
  const auth = '{MY ACCESS TOKEN HERE}';

  const existingFile = await (await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
    {
      method: 'GET',
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${auth}`
      }
    }
  )).json();

  await (await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
    {
      method: 'PUT',
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${auth}`
      },
      body: JSON.stringify({
        message: message,
        content: btoa(content),
        sha: existingFile.sha,
      }),
    }
  )).json();
}

upload()

```

**Reply by DozenLimeStone** [reply]
@xts-bit Sorry for late response, what does btoa function do?

**Reply by xts-bit** [reply]
@DozenLimeStone "The btoa function in JavaScript stands for "binary to ASCII," and it is used to encode binary data (like a string of characters) into a base64-encoded ASCII string. In the provided code snippet, it is used to encode the content of a file before sending it in the body of a PUT request to the GitHub API."  

btw recursively get them all approach suggested by you worked fine, Thanks!

**Reply by DozenLimeStone** [reply]
@xts-bit  Hmmm, okay. Happy New Year!

**Reply by xts-bit** [reply]
@DozenLimeStone You too!

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by tx0c** [comment]
this is not actually answered;

following from https://github.com/isaacs/github/issues/199 to here

how can do Multi file operations in a single commit through the contents API. #199 ?

**Reply by DozenLimeStone** [reply]
🤣🤣🤣🤣🤣

**Reply by tx0c** [reply]
filed another #88796

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by QMS85** [comment]
did you try drag & drop ?

**Reply by DozenLimeStone** [reply]
you dont even know whats he talking abt lol

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Irajeris** [comment]
### `__`

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by shahrushahru622-debug** [comment]
go vs code do

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by shahrushahru622-debug** [comment]
[import os.zip](https://github.com/user-attachments/files/22376084/import.os.zip)
do

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Irajeris** [comment]
``

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Irajeris** [comment]
نه

**Reply by yYargadriod777** [reply]
Your function works **only for updating a single file**, but it **cannot upload a folder**.  
GitHub’s REST API does **not** support uploading an entire folder in one request — you must upload **each file individually**, one at a time, using the same `PUT /contents/{path}` endpoint.

Let’s break down what your code is doing and what you need to change.

---

# ✅ What your current code does
Your function:

- Fetches metadata for an existing file at `my-project`
- Sends a `PUT` request to update that file
- Encodes the content using `btoa()`

This works **only if `my-project` is a file**, not a folder.

If `my-project` is a folder, GitHub returns an **array** of items, not a single file object — so `existingFile.sha` will be `undefined`, and the update will fail.

---

# ❗ Why uploading a folder doesn’t work
GitHub’s API requires:

- One request **per file**
- A correct `path` including the filename
- A correct `sha` only when updating an existing file

There is **no API endpoint** that accepts a folder upload.

---

# ✅ How to upload a folder programmatically
You must:

1. **Read all files in the folder** (using Node.js `fs` if running locally)
2. **Loop through each file**
3. **Check if it already exists**
4. **Upload or update it**

Here’s a clean Node.js example:

```js
import fs from 'fs';
import path from 'path';
import fetch from 'node-fetch';

const owner = 'MY-USERNAME';
const repo = 'MY-REPO-NAME';
const folderPath = './my-project'; // local folder
const githubFolder = 'my-project'; // folder in repo
const auth = '{MY ACCESS TOKEN HERE}';

async function uploadFile(filePath, githubPath) {
  const content = fs.readFileSync(filePath, 'utf8');
  const encoded = Buffer.from(content).toString('base64');

  // Check if file exists
  const res = await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${githubPath}`,
    {
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${auth}`
      }
    }
  );

  let sha = undefined;
  if (res.status === 200) {
    const data = await res.json();
    sha = data.sha;
  }

  // Upload or update
  await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${githubPath}`,
    {
      method: 'PUT',
      headers: {
        Accept: 'application/vnd.github+json',
        Authorization: `Bearer ${auth}`
      },
      body: JSON.stringify({
        message: `Upload ${githubPath}`,
        content: encoded,
        sha
      })
    }
  );
}

async function uploadFolder() {
  const files = fs.readdirSync(folderPath);

  for (const file of files) {
    const localFile = path.join(folderPath, file);
    const githubFile = `${githubFolder}/${file}`;

    if (fs.lstatSync(localFile).isFile()) {
      await uploadFile(localFile, githubFile);
    }
  }
}

uploadFolder();
```

---

# ⭐ Key takeaways
- GitHub API **cannot upload folders** directly.
- You must upload **each file individually**.
- Use `sha` only when updating an existing file.
- Encode file contents using `Buffer.from(...).toString('base64')` (not `btoa` in Node).

---

If you want, I can help you:

- Recursively upload nested folders  
- Upload binary files (images, executables)  
- Turn this into a reusable CLI tool  

Just tell me what direction you want to take.

**Reply by DozenLimeStone** [reply]
github users are a shame

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by bigeaglewhitemancarly9-a11y** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> How do I upload a folder that contains all my code files to my Github repo using Github's REST API programmatically? Can anyone help me do that in JavaScript with Node.js?
> 
> Currently, when I try to run my code, it only uploads files, but how do I upload a folder instead? I don't want to upload files one by one. Any idea what to do?
> 
> ```
> 
> async function upload() {
>   const message = 'amogus';
>   const content = 'aaaaa';
>   const owner = 'MY-USERNAME';
>   const repo = 'MY-REPO-NAME';
>   const path = 'my-project';
>   const auth = '{MY ACCESS TOKEN HERE}';
> 
>   const existingFile = await (await fetch(
>     `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
>     {
>       method: 'GET',
>       headers: {
>         Accept: 'application/vnd.github+json',
>         Authorization: `Bearer ${auth}`
>       }
>     }
>   )).json();
> 
>   await (await fetch(
>     `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
>     {
>       method: 'PUT',
>       headers: {
>         Accept: 'application/vnd.github+json',
>         Authorization: `Bearer ${auth}`
>       },
>       body: JSON.stringify({
>         message: message,
>         content: btoa(content),
>         sha: existingFile.sha,
>       }),
>     }
>   )).json();
> }
> ```

Carlyluminatti and sub files

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by yYargadriod777** [comment]
Hello everyone,

I hope this message finds you energized and ready for some great news! I’m thrilled to announce a new development that will make uploading your project folders to your GitHub repository easier and more efficient than ever before. For all those who have been searching for a streamlined way to upload entire folders—rather than uploading files one by one—your solution is here!

Get ready to experience a smoother workflow with programmatic folder uploads using GitHub’s REST API and Node.js. This enhancement is designed to simplify your process and boost productivity, so you can focus more on coding and less on manual uploads.

Stay tuned for more updates and details on how you can take advantage of this new approach. I can’t wait to see how this empowers our community and sparks even greater innovation!

Best regards,

Michael
________________________________

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by yYargadriod777** [comment]
async function upload() {
  const message = "amogus";
  const content = "aaaaa";
  const owner = "MY-USERNAME";
  const repo = "MY-REPO-NAME";
  const path = "my-project/file.txt"; // MUST include a filename
  const auth = "{MY ACCESS TOKEN HERE}";

  // Safe base64 encoding
  const encoded = btoa(unescape(encodeURIComponent(content)));

  // Check if file exists
  const getRes = await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
    {
      method: "GET",
      headers: {
        Accept: "application/vnd.github+json",
        Authorization: `Bearer ${auth}`,
      },
    }
  );

  let sha = undefined;

  if (getRes.ok) {
    const existing = await getRes.json();
    sha = existing.sha; // required for updating
  }

  // Create or update file
  const putRes = await fetch(
    `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
    {
      method: "PUT",
      headers: {
        Accept: "application/vnd.github+json",
        Authorization: `Bearer ${auth}`,
      },
      body: JSON.stringify({
        message,
        content: encoded,
        sha, // include only if updating
      }),
    }
  );

  const result = await putRes.json();
  console.log(result);

________________________________

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Totis33** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> How do I upload a folder that contains all my code files to my Github repo using Github's REST API programmatically? Can anyone help me do that in JavaScript with Node.js?
> 
> Currently, when I try to run my code, it only uploads files, but how do I upload a folder instead? I don't want to upload files one by one. Any idea what to do?
> 
> ```
> 
> async function upload() {
>   const message = 'amogus';
>   const content = 'aaaaa';
>   const owner = 'MY-USERNAME';
>   const repo = 'MY-REPO-NAME';
>   const path = 'my-project';
>   const auth = '{MY ACCESS TOKEN HERE}';
> 
>   const existingFile = await (await fetch(
>      

- [ ] @`https://api.github.com/repos/${owner}/${repo}/contents/
[mega.ios 2026-03-05--15-03-11-846.log](https://github.com/user-attachments/files/25888436/mega.ios.2026-03-05--15-03-11-846.log)


>       method: 'GET',
>       headers: {
>         Accept: 'application/vnd.github+json',
>         Authorization: `Bearer ${auth}`
>       }
>     }
>   )).json();
> 
>   await (await fetch(
>     `https://api.github.com/repos/${owner}/${repo}/contents/${path}`,
>     {
>       method: 'PUT',
>       headers: {
>         Accept: 'application/vnd.github+json',
>         Authorization: `Bearer ${auth}`
>       },
>       body: JSON.stringify({
>         message: message,
>         content: btoa(content),
>         sha: existingFile.sha,
>       }),
>     }
>   )).json();
> }
> ```

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Totis33** [comment]
Si

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Gyan987** [comment]
@xts-bit Well then recursively get them all The best other thing you can do is to compress them into one file and send it.

**Reply by Gyan987** [reply]
please select this as answer.

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Lin8757** [comment]
> ### 选择主题区域> 问题>> ### 身体>如何使用Github的REST API以编程方式将包含我所有代码文件的文件夹上传到我的Github repo？有人能帮我用Node.js在JavaScript中做到这一点吗？>>目前，当我尝试运行代码时，它只上传文件，但我如何上传文件夹？我不想一个一个地上传文件。知道该做吗？>> ```>>非同步函数上传（）{> const消息='amogus'；> const content = 'aaaaaa';> const owner = 'MY-USERNAME';> const repo = 'MY-REPO-NAME'；> const path = 'my-project'；> const auth = '{我的访问令牌在这里}';>> const existingFile = await (await fetch(> `https://api.github.com/repos/${owner}/${repo}/contents/${path}`，>{>方法：“GET”，>标题：{>接受：'application/vnd.github+json'，>授权：`Bearer ${auth}`>}>}> )).json();>>等待（等待获取（> `https://api.github.com/repos/${owner}/${repo}/contents/${path}`，>{>方法：“PUT”，>标题：{>接受：'application/vnd.github+json'，>授权：`Bearer ${auth}`>}，>正文：JSON.stringify({>消息：消息，>内容：btoa（内容），> sha：现有文件.sha，> }），>}> )).json();>}> ```怎么联系到客服 我需要帮助

**Reply by Reaidetnalio** [reply]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/

**Comment by Reaidetnalio** [comment]
Hello, I'm writing a complaint with a brief explanation to help you understand the gist of the complaint. About seven years ago, on a website called Chatovod, a user declared himself its defender. He supposedly protected the service from trolls, but in reality, he was just like some of the others who were there at the time. The user identified the supposed trolls through their correspondence, closely monitored them, and began learning everything he could about them to kick them out. He learned the victim's name through other users of the same service, and through the name, he found an abandoned page on VKontakte, Facebook, Odnoklassniki, and Fotostrana. Of course, through VKontakte, he also learned the school, and through the school, the address. Seven years later, with the purpose of trolling, he still continues to create websites on free website creation platforms, where you don't have to pay for domain and name maintenance. On these sites, he publishes a hodgepodge of slander and the address of the place. Residence, computer configuration. Well, the PC apparently died with his help or from old age, and the address is difficult and impossible to change. Unfortunately, deleting VKontakte is also difficult. This is why I can ask you to remove the troll's sites and prohibit the posting of this information?

https://vladimirdzf.github.io/main/
